beacon_node/beacon_chain/src/beacon_chain.rs

```diff
@@ -1976,10 +1976,22 @@ impl<T: BeaconChainTypes> BeaconChain<T> {
 
         drop(validator_monitor);
 
-        metrics::observe(
-            &metrics::OPERATIONS_PER_BLOCK_ATTESTATION,
-            block.body().attestations().len() as f64,
-        );
+        // Only present some metrics for blocks from the previous epoch or later.
+        //
+        // This helps avoid noise in the metrics during sync.
+        if block.slot().epoch(T::EthSpec::slots_per_epoch()) + 1 >= self.epoch()? {
+            metrics::observe(
+                &metrics::OPERATIONS_PER_BLOCK_ATTESTATION,
+                block.body().attestations().len() as f64,
+            );
+
+            if let Ok(block) = block.as_altair() {
```

Comment:
This would be good as `block.body().sync_aggregate()` so that it works for all post-Altair block types, but I realise that the `sync_aggregate` method doesn't exist yet in `unstable`.

When we implement the next fork we should audit all uses of `as_altair` anyway, so I'm happy to merge without fixing this.

The definition of `sync_aggregate` is:

```rust
impl<'a, T: EthSpec> BeaconBlockBodyRef<'a, T> {
    /// Access the sync aggregate from the block's body, if one exists.
    pub fn sync_aggregate(self) -> Option<&'a SyncAggregate<T>> {
        match self {
            BeaconBlockBodyRef::Base(_) => None,
            BeaconBlockBodyRef::Altair(inner) => Some(&inner.sync_aggregate),
        }
    }
}
```

in `beacon_block_body.rs`

Reply:
Oh, great idea. I hadn't considered this problem before. I've added this function in https://github.com/sigp/lighthouse/pull/2439/commits/0dc02e564e4385d36e76e36270984ffe17bc173c. Thanks for laying it all out nicely for me :slightly_smiling_face:

Reply:
Noice :sunglasses: